deepchem/feat/rdkit_grid_featurizer.py

```diff
@@ -382,71 +377,60 @@ def featurize_splif(protein_xyz, protein, ligand_xyz, ligand, contact_bins,
   return (splif_dicts)
 
 
-def compute_ring_center(mol, ring):
-  ring_xyz = np.zeros((len(ring._path), 3))
-  for i, atom_idx in enumerate(ring._path):
-    atom = mol.GetAtom(int(atom_idx))
-    ring_xyz[i, :] = [atom.x(), atom.y(), atom.z()]
+def compute_ring_center(mol, ring_indices):
+  conformer = mol.GetConformer()
+  ring_xyz = np.zeros((len(ring_indices), 3))
+  for i, atom_idx in enumerate(ring_indices):
+    atom_position = conformer.GetAtomPosition(atom_idx)
+    ring_xyz[i] = np.array(atom_position)
   ring_centroid = compute_centroid(ring_xyz)
   return ring_centroid
 
 
-def compute_ring_normal(mol, ring):
+def compute_ring_normal(mol, ring_indices):
+  conformer = mol.GetConformer()
   points = np.zeros((3, 3))
-  for i, atom_idx in enumerate(ring._path):
-    if i == 3: break
-    atom = mol.GetAtom(int(atom_idx))
-    points[i, :] = [atom.x(), atom.y(), atom.z()]
+  for i, atom_idx in enumerate(ring_indices[:3]):
+    atom_position = conformer.GetAtomPosition(atom_idx)
+    points[i] = np.array(atom_position)
 
-  v1 = points[1, :] - points[0, :]
-  v2 = points[2, :] - points[0, :]
+  v1 = points[1] - points[0]
+  v2 = points[2] - points[0]
   normal = np.cross(v1, v2)
   return normal
 
 
-def is_pi_parallel(protein_ring_center, protein_ring_normal, ligand_ring_center,
-                   ligand_ring_normal):
-  dist = np.linalg.norm(protein_ring_center - ligand_ring_center)
-  angle = angle_between(protein_ring_normal, ligand_ring_normal) * 180 / np.pi
-  if ((np.abs(angle) < 30.0) or
-      (np.abs(angle) > 150.0 and np.abs(angle) < 210.0) or
-      (np.abs(angle) > 330.0 and np.abs(angle) < 360.0)):
-    if dist < 8.0:
-      return True
-
-  return False
-
-
-def is_pi_t(protein_ring_center, protein_ring_normal, ligand_ring_center,
-            ligand_ring_normal):
-  dist = np.linalg.norm(protein_ring_center - ligand_ring_center)
-  angle = angle_between(protein_ring_normal, ligand_ring_normal) * 180 / np.pi
-  if ((np.abs(angle) > 60.0 and np.abs(angle) < 120.0) or
-      (np.abs(angle) > 240.0 and np.abs(angle) < 300.0)):
-    if dist < 5.5:
-      return True
-
-  return False
-
-
-def update_feature_dict(feature_dict, idxs=None, indices=None):
-  if idxs is not None:
-    indices = []
-    for idx in idxs:
-      indices.append(idx - 1)
+def is_pi_parallel(ring1_center,
+                   ring1_normal,
+                   ring2_center,
+                   ring2_normal,
+                   dist_cutoff=8.0,
+                   angle_cutoff=30.0):
+  dist = np.linalg.norm(ring1_center - ring2_center)
+  angle = angle_between(ring1_normal, ring2_normal) * 180 / np.pi
+  if ((angle < angle_cutoff or angle > 180.0 - angle_cutoff) and
```

Comment:
Recommend just directly returning the boolean value here (the if/else isn't needed)

Reply:
I think complex return statements are considered a bad practice in python, especially that long.

Reply:
I side with @mwojcikowski  on this one.  Also you can remove the else as it is implied by returning True on the other path.

Reply:
I bow to the wisdom of the crowd :+1:

Reply:
OK, then I'll remove the else and leave the if-return part as it is